js/src/sdk/utils/errors/src/formatter.ts

```diff
@@ -15,72 +21,56 @@ interface ErrorDetails {
 }
 
 export const getAPIErrorDetails = (
-  errorKey: string,
-  axiosError: AxiosError<ErrorResponseData>,
-  predefinedError: Record<string, unknown>
+  axiosError: AxiosError<ErrorResponseData>
 ): ErrorDetails => {
+  const statusCode = axiosError.response?.status;
+  const errorCode = statusCode
+    ? API_TO_SDK_ERROR_CODE[statusCode]
+    : COMPOSIO_SDK_ERROR_CODES.BACKEND.UNKNOWN;
+  const predefinedError = BASE_ERROR_CODE_INFO[errorCode];
+
   const defaultErrorDetails = {
     message: axiosError.message,
     description:
-      axiosError.response?.data?.message ||
+      axiosError.response?.data?.error?.message ||
       axiosError.response?.data?.error ||
       axiosError.message,
     possibleFix:
       "Please check the network connection, request parameters, and ensure the API endpoint is correct.",
   };
 
   const metadata = generateMetadataFromAxiosError(axiosError);
-  switch (errorKey) {
+
+  const errorNameFromBE = axiosError.response?.data?.error?.name;
+  const errorTypeFromBE = axiosError.response?.data?.error?.type;
+  const errorMessage = axiosError.response?.data?.error?.message;
+
+  const genericMessage = `${errorNameFromBE || predefinedError.message} ${errorTypeFromBE ? `- ${errorTypeFromBE}` : ""} on ${axiosError.config?.baseURL! + axiosError.config?.url!}`;
```

Comment:
Consider adding null/undefined checks for `axiosError.config?.baseURL` and `axiosError.config?.url`. While the optional chaining helps prevent crashes, concatenating with undefined values could lead to "undefined" appearing in the error message.